electrum/gui/qml/qedaemon.py

```diff
@@ -149,6 +151,9 @@ class QEDaemon(AuthMixin, QObject):
 
     def __init__(self, daemon: 'Daemon', plugins: 'Plugins', parent=None):
         super().__init__(parent)
+        if QEDaemon.instance:
+            raise RuntimeError('There should only be one QEDaemon instance')
+        QEDaemon.instance = self
```

Comment:
In general I think we should only introduce singletons sparingly. E.g. they can make testing significantly more complicated.
However if I understood you correctly, in [another thread](https://github.com/spesmilo/electrum/pull/9705#discussion_r2031486360), you said these are already effectively singletons in the QML context? In which case I guess we might as well make them singletons in python too.

Reply:
The `QE*` variants are singletons, the underlying `SimpleConfig` and `Daemon` not necessarily, but the QML UI uses a single `QEDaemon` and a single `QEConfig`. The backend wallet could well remain non-singleton and allow flexibility, although I think the backend also mostly assumes a single daemon and passes around the same SimpleConfig instance everywhere.

Reply:
note: in the non-GUI code, atm I think only the Network object is a singleton (and even that I don't really like! :P).
For a while the SimpleConfig used to be a singleton as well, but we managed to untangle that, and the unit tests in particular are now creating configs left-and-right :P

Oh, the asyncio_loop is also a singleton. That is probably also ok, as having multiple asyncio event loops is a bug magnet AFAIU.

But again, if the claim is that QEDaemon and QEConfig were already effectively singletons in QML just not in python, maybe it's fine to make it homogeneous.